File: Frontend/bytejack_frontend/src/components/Player.js
import Hand from "./Hand";
import { cardCount } from "../utils/utils";

const Player = ({ name, bet, hand, chips, state}) => {
  return (
    <div className="padding">
       <div id="bet">
        {state}
        <span></span>
      </div>
      <span id="name">
        Name: {name}
        <span></span>
      </span>
      <div id="bet">
        Bet: ${bet}
        <span></span>
      </div>
      <div id="bet">
        Chips: ${chips}
        <span></span>
      </div>
      <div id="count">
        Card Count: {cardCount(hand)}
        <span></span>
      </div>
      <Hand cards={hand}></Hand>
    </div>
  );
};

export default Player;

File: backend/player.py
from hand import Hand

class Player():
    def __init__(self, name, id, chips):
        self.name = name
        self.id = id
        self.hand = Hand()
        self.bet = 0
        self.chips = chips

    def get_name(self):
        return self.name
    
    def get_id(self):
        return self.id

    def get_bet(self):
        return self.bet
    
    def set_bet(self, amount):
        if type(amount) == type("example"):
            amount = int(amount)
        self.bet = amount
    
    def get_chips(self):
        return self.chips
    
    def hand_total(self):
        return self.hand.get_hand_total()
        
    def get_hand(self):
        return self.hand.get_hand()
    
    def add_card(self, card):
        self.hand.add_card(card)

    def clear_hand(self):
        self.hand.clear_hand()

    def subtract_chips(self, num):
        newNum = int(num)
        self.chips -= newNum

    def add_chips(self, num):
        newNum = int(num)
        self.chips += newNum #new amoiunt in front of old amount
        print(self.chips)